atc/api/accessor/accessor.go

```diff
@@ -135,18 +135,18 @@ func (a *access) rolesForTeam(auth atc.TeamAuth) []string {
 
 		// backwards compatibility for allow-all-users
 		if len(userAuth) == 0 && len(groupAuth) == 0 {
-			roles = append(roles, role)
+			roleSet[role] = true
```

Comment:
it seems like we're not testing this anywhere - should we have a test that covers it?

Reply:
nah this is legacy